arithmetization/src/test/java/net/consensys/linea/zktracer/precompiles/BlakeTests.java

```diff
@@ -0,0 +1,119 @@
+import static net.consensys.linea.zktracer.module.blake2fmodexpdata.Trace.INDEX_MAX_BLAKE_RESULT;
+import static net.consensys.linea.zktracer.module.constants.GlobalConstants.LLARGE;
+
+import net.consensys.linea.testing.BytecodeCompiler;
+import net.consensys.linea.testing.BytecodeRunner;
+import net.consensys.linea.zktracer.opcode.OpCode;
+import org.apache.tuweni.bytes.Bytes;
+import org.hyperledger.besu.datatypes.Address;
+import org.junit.jupiter.api.Test;
+
+public class BlakeTests {
+  @Test
+  void emptyBlakeTest() {
+    final Bytes bytecode =
+        BytecodeCompiler.newProgram()
+            .push(0)
+            .push(0)
+            .push(0)
+            .push(0)
+            .push(0)
+            .push(Address.BLAKE2B_F_COMPRESSION) // address
+            .push(0xffff) // gas
+            .op(OpCode.CALL)
+            .op(OpCode.POP)
+            .compile();
+    BytecodeRunner.of(bytecode).run();
+  }
+
+  @Test
+  void basicBlakeTest() {
+    final Bytes bytecode =
+        BytecodeCompiler.newProgram()
+            .push(Bytes.fromHexString("0x0badb077")) // value, some random data to hash
+            .push(5) // offset
+            .op(OpCode.MSTORE)
+            .push(10) // value = r for Blake call
+            .push(3) // offset
+            .op(OpCode.MSTORE8)
+            .push(BLAKE2f_HASH_OUTPUT_SIZE) // return size
+            .push(0) // return offset
+            .push(213) // size
+            .push(0) // offset
+            .push(0) // value
+            .push(Address.BLAKE2B_F_COMPRESSION) // address
+            .push(0xffff) // gas
+            .op(OpCode.CALL)
+            .op(OpCode.POP)
+            // To check memory consistency
+            .push(0)
+            .op(OpCode.MLOAD)
+            .op(OpCode.STOP)
+            .compile();
+    BytecodeRunner.of(bytecode).run();
+  }
+
+  @Test
+  void wrongFInputTest() {
+    final Bytes bytecode =
+        BytecodeCompiler.newProgram()
+            .push(Bytes.fromHexString("0x0badb077")) // value, some random data to hash
+            .push(5) // offset
+            .op(OpCode.MSTORE)
+            .push(2) // value = f for Blake call
+            .push(212) // offset
+            .op(OpCode.MSTORE8)
+            .push(LLARGE * (INDEX_MAX_BLAKE_RESULT + 1)) // return size
+            .push(0) // return offset
+            .push(213) // size
+            .push(0) // offset
+            .push(0) // value
+            .push(Address.BLAKE2B_F_COMPRESSION) // address
+            .push(0xffff) // gas
+            .op(OpCode.CALL)
+            .op(OpCode.POP)
+            .compile();
+    BytecodeRunner.of(bytecode).run();
+  }
+
+  @Test
+  void notEnoughGasBlakeTest() {
+    final Bytes bytecode =
+        BytecodeCompiler.newProgram()
+            .push(Bytes.fromHexString("0x0badb077")) // value, some random data to hash
+            .push(5) // offset
```

Comment:
What is the intent here. You are producing this in memory:

```
<-- 5 bytes -->    <--      32 bytes       -->
00 00 00 00 00  |  00 00 00 ... 00 0b ad b0 77
```

Reply:
yes, I just want to have something (ie  not 0) in the hash input data, so really don't care of what I put inside as soon as we have a non zero between offset 5 and 212.